packages/next/src/build/output/log.ts

```diff
@@ -79,6 +79,10 @@ export function trace(...message: any[]) {
 const warnOnceMessages = new Set()
 export function warnOnce(...message: any[]) {
   if (!warnOnceMessages.has(message[0])) {
+    if (warnOnceMessages.size > 1000) {
+      // delete the oldest message (FIFO) to prevent memory leak
+      warnOnceMessages.delete(warnOnceMessages.values().next().value)
```

Comment:
We could use the LRU handler we have here https://github.com/vercel/next.js/blob/canary/packages/next/src/server/lib/lru-cache.ts

Reply:
Good idea! Added in [5f598cb](https://github.com/vercel/next.js/pull/73483/commits/5f598cb473d794df4e42c41511a992cf013ecc12)